### PR TITLE
Misc browse window functions

### PR DESCRIPTION
**Pull request by AaronVanGeffen**

### CODE REVIEW

File: src/openloco/windows/promptbrowsewnd.cpp [hunk 1]
```diff
@@ -733,21 +733,43 @@ namespace openloco::ui::prompt_browse
...
     static void sub_446574(ui::window* window)
     {
         if (*_type != browse_file_type::unk_2)
         {
-            call(0x00446689);
-        }
-        else
-        {
-            if (sub_446F1D())
+            if (filenameContainsInvalidChars())
```
Comment (AaronVanGeffen) [inline src/openloco/windows/promptbrowsewnd.cpp:772]: This change is in line with the disassembly, I think, however it seems to be skipping the filename check for type 0 and 1 (save game, landscape). This doesn't seem appropriate. Can someone verify?

Comment (marijnvdwerf) [reply src/openloco/windows/promptbrowsewnd.cpp:772]: `_type` is a `browse_type`, not a `browse_file_type`.

Comment (marijnvdwerf) [reply src/openloco/windows/promptbrowsewnd.cpp:772]: Some structure akin to the following would make more sense to me.
```
if (_type == save) {
   validateFileName();
   // ...
}

sub_446689();
```

File: src/openloco/windows/promptbrowsewnd.cpp [hunk 2]
```diff
@@ -733,21 +732,43 @@ namespace openloco::ui::prompt_browse
             });
     }
 
-    static int sub_446F1D()
+    // 0x00446F1D
+    static bool filenameContainsInvalidChars()
     {
-        return call(0x00446F1D) & X86_FLAG_CARRY;
+        uint8_t numNonSpacesProcessed = 0;
+        for (char* ptr = &*_text_input_buffer; *ptr != '\0'; ptr++)
```
Comment (duncanspumpkin) [inline src/openloco/windows/promptbrowsewnd.cpp:739]: `const char *`

the &* looks horrible but I know its just cause of the loco_global.

Comment (AaronVanGeffen) [reply src/openloco/windows/promptbrowsewnd.cpp]: Yeah, we should probably just redefine the `&` operator for `loco_global` as well.

**Pull request merged**